MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpressBlockIo.c

```diff
@@ -169,6 +169,13 @@ NvmeRead (
   BOOLEAN                       IsEmpty;
   EFI_TPL                       OldTpl;
 
+  // MU_CHANGE [BEGIN] - Request Number of Queues from Controller
+  if (&Device->AsyncQueue == NULL) {

```

Comment:
This can never be `NULL`, can it? AsyncQueue is a LIST_ENTRY, so the address of it would be in the Device struct? Should we be doing `IsListEmpty` instead?

Reply:
In the existing code they raise the TPL before checking if the AsyncQueue is empty, should we not check that and instead check about the number of queues? Or just take the TPL raise/lower hit and check the list there?

Reply:
This was an artifact of a bug that was introduced and then fixed in this PR. It led to dereferencing a NULL pointer. That's been fixed by always initializing the async list. Good catch, I removed these now